### PR TITLE
feat: Support glob paths in file plugin `update` extra

### LINKED ISSUE #6585: feature: support glob patterns for paths in file bundle plugin `update` extra
### Feature scope

CLI (options, error messages, logging, etc.)

### Overview
File bundle plugins can specify a number of files to update with the `update` extra, when `meltano upgrade files` is run.

`meltano.yml`
```yml
version: 1
default_environment: dev
environments:
- name: dev
- name: staging
- name: prod
project_id: fefc3baf-ebb0-4f68-87d1-fe5b3afbe6e8
plugins:
  files:
  - name: files-dbt
    pip_url: git+https://github.com/meltano/files-dbt
    update:
      transform/models/.gitkeep: true
      transform/profile/profiles.yml: true
      transform/.gitignore: true
      transform/dbt_project.yml: true
```

Currently, each file than can be updated by `meltano upgrade files` must have its path declared under `update` individually. This can lead to bloated `meltano.yml` file bundle definitions that specify many files within common directories as upgradable.

### Proposal
Support glob patterns to reduce the number of paths needed to specify all files required for upgrade.

All bundle files:
```yml
    update:
      '**/*': true
```

All bundle `.yml` files:
```yml
    update:
      '**/*.yml': true
```

All bundle `.yml` files under the `transform` directory:
```yml
    update:
      'transform/**/*.yml': true
```

### PR DESCRIPTION
**Pull request by ReubenFrankel**
Resolves #6585 to support glob paths in file plugin `update` extra.

---

### Checklist

- [x] Implement feature
- [x]  Add test(s)
- [x]  Update docs

### CODE REVIEW

**Comment by ReubenFrankel** [comment]
[PurePath.match](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.match) behaves a little differently to [Path.glob](https://docs.python.org/3/library/pathlib.html#pathlib.Path.glob), and so doesn't currently support `**` (this directory and all subdirectories, recursively). See:
- https://stackoverflow.com/questions/67827443/python-pathlib-match-function-doesnt-work
- https://discuss.python.org/t/pathlib-path-match-and/9052
- https://bugs.python.org/issue29249

**Review by edgarrmondragon — APPROVED** [review]
LGTM! Thanks @ReubenFrankel 🙂

File: tests/meltano/cli/test_upgrade.py [hunk 1]
```diff
@@ -155,6 +156,47 @@ def test_upgrade_files(
...
+                "config",
+                "--plugin-type",
+                "files",
+                "airflow",
+                "set",
+                "_update",
+                json.dumps({
+                    "orchestrate/dags/meltano.py": False,
+                    "*.py": True,
+                })
```
Comment (edgarrmondragon) [inline tests/meltano/cli/test_upgrade.py:190]:
```suggestion
                json.dumps(
                    {
                        "orchestrate/dags/meltano.py": False,
                        "*.py": True,
                    },
                ),
```

**Review by tayloramurphy — APPROVED** [review]
Docs approval 🚢

**Comment by edgarrmondragon** [comment]
Blocked by https://github.com/meltano/meltano/pull/6637

**Comment by edgarrmondragon** [comment]
@ReubenFrankel I've fixed the remaining linter issues. Thanks!

**Pull request merged**